src/libs/Network/NetworkStore.ts

```diff
@@ -82,85 +78,51 @@ Onyx.connect({
             triggerReconnectCallback();
         }
 
-        offline = Boolean(network.shouldForceOffline) || network.isOffline;
+        offline = Boolean(network.shouldForceOffline) || !!network.isOffline;
     },
 });
 
-/**
- * @returns {Object}
- */
-function getCredentials() {
+function getCredentials(): Credentials | null {
     return credentials;
 }
 
-/**
- * @returns {Boolean}
- */
-function isOffline() {
+function isOffline(): boolean {
     return offline;
 }
 
-/**
- * @returns {String}
- */
-function getAuthToken() {
+function getAuthToken(): string | null {
```

Comment:
Hmmm should we make authToken always a string and have it be the empty string when it is not set?

Reply:
I believe `null` is a better value to signify that `authToken` doesn't exist, empty string can be tricky sometimes. In javascript I would agree with you to make it an empty string but with TS in mind I'm inclined to null.

Reply:
Interesting, can you delve a bit more why in JS yes but in TS no?

Reply:
In JavaScript, both null and an empty string are considered "falsy" values. This means that in a conditional statement, they'll both be treated as `false`. So there's little practical difference to use either of them to signify the absence of a value.

TypeScript has strict type checking and treats `null` and an empty string differently. By using `null` in TypeScript, we add more semantic meaning to our code. A `null` authToken indicates that the token does not exist or it hasn't yet been set, while an empty string might unintentionally suggest that the token is present but with a "" value, which could lead to confusion.